src/app/app-routing.module.ts

```diff
@@ -2,24 +2,22 @@ import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
 import { HomeComponent } from './home/home.component';
-import { AdminAPIComponent } from './admin-api/admin-api.component';
-import { Auth, PermissionGuard } from "./_utils"
+import { Auth } from "./_utils"
+
 
-//redirect
 const routes: Routes = [
   {
-    path: 'admin', children: [
-      { path: "", component: AdminAPIComponent, pathMatch: "full" },
-      { path: ':id', component: AdminAPIComponent, canActivate: [PermissionGuard] },
-    ],
+    path: 'admin',
+    loadChildren: () => import('./modules/admin-api/admin-api.module').then(m => m.AdminAPIModule),
     canActivate: [Auth]
```

Comment:
let's try to explore what types of guards are there.

Current behavior is that even though my user is not authorized - chunk with AdminAPIModule will be still loaded to user's browser

let's pretend that we would like to deal with this situation as we're bulding some app that will be used by mobile users with bad internet.
Thus, let's use some different guard type that will be deciding if children can be loaded